torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -42,13 +42,10 @@ struct TORCH_API AccumulateGrad : public Node {
 
   variable_list apply(variable_list&& grads) override;
 
-  static at::Tensor callHooks(const Variable& variable, at::Tensor new_grad) {
-    for (auto& hook : impl::hooks(variable)) {
-      new_grad = (*hook)({new_grad})[0];
-    }
-    return new_grad;
+  std::vector<std::unique_ptr<FunctionPreHook>>& tensor_pre_hooks() noexcept
```

Comment:
nit could you leave a commend here that says:
Since the AccumulateGrad Node is only a weak ref from the Tensor, it can be destroyed even though the Tensor is still alive (contrary to all other Nodes). So we must lazily read the Tensor hooks here.